File: algorithm/typescript/src/contains_duplicate.test.ts
import { contains_duplicate } from "./contains_duplicate";

test("test contains_duplicates", () => {
  let testCases: [number[], boolean][] = [
    [[], false],
    [[1, 2, 3, 4, 1], true],
    [[1, 2, 3, 4, 5], false],
    [[1, 2, 1, 2, 5], true],
  ];

  testCases.forEach(([input, expected]) => {
    expect(contains_duplicate(input)).toEqual(expected);
  });
});

File: algorithm/kotlin/lib/src/test/kotlin/com/hungknow/ContainsDuplicateTest.kt
package com.hungknow.kotlin

import kotlin.test.Test
import kotlin.test.assertEquals

class ContainsDuplicateTest {
    @Test
    fun containsDuplicate() {
        val testcases =
                arrayOf(
                        intArrayOf(1, 2, 3, 1) to true,
                        intArrayOf(1, 2, 3, 4) to false,
                        intArrayOf(11, 1, 1, 3, 3, 4, 3, 2, 4, 2) to true
                )
        val containsDuplicate = ContainsDuplicate()
        for (testcase in testcases) {
            val input = testcase.first
            val expected = testcase.second
            assertEquals(expected, containsDuplicate.containsDuplicate(input))
            assertEquals(expected, containsDuplicate.containsDuplicate2(input))
            assertEquals(expected, containsDuplicate.containsDuplicate3(input))
        }
    }
}
